test/dynamo/test_higher_order_ops.py

```diff
@@ -3426,14 +3607,7 @@ def wrapper_fn(x, y):
 
         actual = wrapper_fn(x, y)
         expected = torch.compile(wrapper_fn, backend="aot_eager", fullgraph=False)(x, y)
-        self.assertEqual(len(counters["graph_break"]), 1)
-        assert_dict_matches_regex(
-            self,
-            dict(counters["graph_break"]),
-            {
-                r".*HigherOrderOperator: Mutating a variable not in the current scope \(SideEffects\)": 2
-            },
-        )
+        self.assertEqual(len(counters["graph_break"]), 0)
```

Comment:
Can we check that some_list == [1, 1]? 

Also, can we add a test for what happens if the function looks like:

```
def f(x, y):
   some_list.append(x)
   return x + y
```

Reply:
This one doesn't work

```
Traceback (most recent call last):
  File "/home/guilhermeleobas/micromamba/envs/pytorch/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "test/dynamo/test_higher_order_ops.py", line 3631, in test_vmap_side_effects_append_input
    expected = torch.compile(wrapper_fn, backend="aot_eager", fullgraph=False)(x, y)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/eval_frame.py", line 417, in _fn
    return fn(*args, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/eval_frame.py", line 580, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/convert_frame.py", line 741, in _convert_frame
    result = inner_convert(frame, cache_entry, hooks, frame_state)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/convert_frame.py", line 384, in _convert_frame_assert
    return _compile(
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/convert_frame.py", line 643, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/utils.py", line 247, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/convert_frame.py", line 524, in compile_inner
    out_code = transform_code_object(code, transform)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/bytecode_transformation.py", line 1033, in transform_code_object
    transformations(instructions, code_options)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/convert_frame.py", line 151, in _fn
    return fn(*args, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/convert_frame.py", line 489, in transform
    tracer.run()
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/symbolic_convert.py", line 2111, in run
    super().run()
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/symbolic_convert.py", line 780, in run
    and self.step()
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/symbolic_convert.py", line 743, in step
    getattr(self, inst.opname)(inst)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/symbolic_convert.py", line 2229, in RETURN_VALUE
    self.output.compile_subgraph(
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/output_graph.py", line 940, in compile_subgraph
    self.compile_and_call_fx_graph(tx, pass2.graph_output_vars(), root)
  File "/home/guilhermeleobas/micromamba/envs/pytorch/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/output_graph.py", line 1085, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/utils.py", line 247, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/output_graph.py", line 1157, in call_user_compiler
    raise BackendCompilerFailed(self.compiler_fn, e).with_traceback(
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/output_graph.py", line 1138, in call_user_compiler
    compiled_fn = compiler_fn(gm, self.example_inputs())
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/repro/after_dynamo.py", line 117, in debug_wrapper
    compiled_gm = compiler_fn(gm, example_inputs)
  File "/home/guilhermeleobas/git/pytorch/torch/__init__.py", line 1753, in __call__
    return self.compiler_fn(model_, inputs_, **self.kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/backends/common.py", line 55, in compiler_fn
    cg = aot_module_simplified(gm, example_inputs, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_functorch/aot_autograd.py", line 889, in aot_module_simplified
    compiled_fn = create_aot_dispatcher_function(
  File "/home/guilhermeleobas/git/pytorch/torch/_dynamo/utils.py", line 247, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/guilhermeleobas/git/pytorch/torch/_functorch/aot_autograd.py", line 507, in create_aot_dispatcher_function
    fw_metadata = run_functionalized_fw_and_collect_metadata(
  File "/home/guilhermeleobas/git/pytorch/torch/_functorch/_aot_autograd/collect_metadata_analysis.py", line 217, in inner
    curr_storage = StorageWeakRef(o.untyped_storage())
torch._dynamo.exc.BackendCompilerFailed: backend='aot_eager' raised:
NotImplementedError: Cannot access storage of BatchedTensorImpl
```

Reply:
Anyway, I added a test for this case but decorated it with `@unittest.expectedFailure`

Reply:
Yeah that's fine